task/host-sp-comms/src/main.rs

```diff
@@ -658,11 +676,35 @@ impl ServerImpl {
     }
 }
 
+impl idl::InOrderHostSpCommsImpl for ServerImpl {
+    fn set_status(
+        &mut self,
+        _msg: &userlib::RecvMessage,
+        status: u64,
+    ) -> Result<(), RequestError<HostSpCommsError>> {
+        let status = Status::from_bits(status).ok_or_else(|| {
+            RequestError::from(HostSpCommsError::InvalidStatus)
+        })?;
```

Comment:
Nit: because `RequestError<T>` implements `From<T>`, and `?` automatically calls `from`, you can do
```suggestion
        let status =
            Status::from_bits(status).ok_or(HostSpCommsError::InvalidStatus)?;
```